File: easy/1_fizz_buzz.py
import sys

with open(sys.argv[1]) as f:
    for line in f:
        line = line.strip()
        if not line:
            continue

        X, Y, n = (int(x) for x in line.split())
        for i in range(1, n+1):
            fizz = (i % X == 0)
            buzz = (i % Y == 0)

            if i > 1:
                sys.stdout.write(' ')
            if fizz:
                sys.stdout.write('F')
            if buzz:
                sys.stdout.write('B')
            if not (fizz or buzz):
                sys.stdout.write(str(i))
        print('')

File: easy/1_fizz_buzz.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <ctype.h>
#include <stdbool.h>

int main(int argc, const char * argv[]) {
    FILE *file = fopen(argv[1], "r");
    char line[1024];
    int x, y, n;
    bool fizz, buzz;

    while (fgets(line, 1024, file)) {
        sscanf(line, "%d %d %d\n", &x, &y, &n);
        for (int i=1; i<=n; i++) {
            fizz = (i % x == 0);
            buzz = (i % y == 0);

            if (i > 1)
                printf(" ");
            if (fizz)
                printf("F");
            if (buzz)
                printf("B");
            if (!(fizz || buzz))
                printf("%d", i);
        }
        printf("\n");
    }

    return 0;
}
